tasks/insights-client-unregistration.yml

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Call insights status
+  include_tasks: check-insights-status.yml
+  when: rhc_insights.state | d("present") == "absent"
+
+- name: Unregister insights-client
+  command: insights-client --unregister
+  when:
+    - '"insights-client" in ansible_facts.packages'
+    - rhc_insights.state | d("present") == "absent"
+    - '"This host is registered" in __rhc_insights_status.stdout'
+    - '"Registered" in __rhc_insights_status.stdout'
+  register: __rhc_insights_unregister_result
```

Comment:
I think we can use a `block` for both, factorizing some of the conditions:
```yaml
- when:
    - rhc_insights.state | d("present") == "absent"
  block:
    ...
```
Also, adding the `'"insights-client" in ansible_facts.packages'` condition in that block (used by the "Unregister insights-client" task) allows you to remove it from `check-insights-status.yml` (when used again later in `insights-client.yml`, it is executed after `insights-client` is installed, so it will be definitely present).

Reply:
Good one! I'm adding blocks to the main tasks